src/cogent3/core/new_alignment.py

```diff
@@ -138,11 +142,73 @@ def copy(self, sliced: bool = False):
         return self
 
 
-class SeqsData:
+class SeqsDataABC(ABC):
+    """Abstract base class for respresenting the collection of sequences underlying
+    a SequenceCollection
+    """
+
+    alphabet: new_alpha.CharAlphabet
+    make_seq: Callable = None
```

Comment:
add a comment about refactoring the type hint for callable. (At some point we should be specific in the type hints for the input argument types about and the returned value type.)